src/main/java/it/gov/pagopa/payhub/auth/service/a2a/ClientServiceImpl.java

```diff
@@ -25,4 +35,19 @@ public ClientDTO registerClient(String clientName, String organizationIpaCode) {
 		Client client = clientRegistrationService.registerClient(clientName, organizationIpaCode);
 		return clientMapper.mapToDTO(client);
 	}
+
+	@Override
+	public String getClientSecret(String organizationIpaCode, String clientId) {
+		byte[] clientSecret = clientRetrieverService.getClientSecret(organizationIpaCode, clientId);
+		return dataCipherService.decrypt(clientSecret);
```

Comment:
the decrypt operation its a retriever op... not something done by the this service...

Reply:
I'll move it in the clientRetrieverService

Reply:
in https://github.com/pagopa/p4pa-auth/pull/86